backend/graphql/types/courseType.ts

```diff
@@ -1,13 +1,33 @@
 import { gql } from "apollo-server-express";
 
 const courseType = gql`
+  type ModuleDTO {
+    id: ID!
+    title: String!
+    description: String
+    image: String
+    previewImage: String
+    published: Boolean!
+    lessons: [String]
+  }
+
+  input ModuleInputDTO {
+    id: ID
+    title: String!
+    description: String
+    image: String
+    previewImage: String
+    published: Boolean!
+    lessons: [String]
+  }
+
   type CourseResponseDTO {
     id: ID!
     title: String!
     description: String
     image: String
     previewImage: String
-    lessons: [ID!]!
+    modules: [ModuleDTO]
     private: Boolean!
     published: Boolean!
   }
```

Comment:
same comment as on `ModuleDTO`, if it's a response I think they should always be required